src/main/java/io/antmedia/datastore/db/MapDBStore.java

```diff
@@ -1070,6 +1076,160 @@ public boolean revokeTokens(String streamId) {
 		return listToken;
 	}
 
+	@Override
+	public List<Subscriber> listAllSubscribers(String streamId, int offset, int size) {
+		List<Subscriber> list = new ArrayList<>();
+		List<Subscriber> listSubscriber = new ArrayList<>();
+
+		synchronized (this) {
+			Collection<String> values = subscriberMap.values();
+			int t = 0;
+			int itemCount = 0;
+			if (size > MAX_ITEM_IN_ONE_LIST) {
+				size = MAX_ITEM_IN_ONE_LIST;
+			}
+			if (offset < 0) {
+				offset = 0;
+			}
+
+			Iterator<String> iterator = values.iterator();
+
+			while(iterator.hasNext()) {
```

Comment:
In some cases MapDBStore is inconsistent. We have protected the map by length. Check the getVodList method